sky/clouds/service_catalog/config.py

```diff
@@ -0,0 +1,34 @@
+"""Configuration for service catalog."""
+
+import contextlib
+import threading
+
+from sky.utils import common_utils
+
+local_config = threading.local()
+# Whether the caller requires the catalog to be narrowed down
+# to the account-specific catalog or just the raw catalog
+# fetched from SkyPilot catalog service.
+local_config.use_default_catalog = False
+
+
+@contextlib.contextmanager
+def set_use_default_catalog(name: str, to_use: bool):
```

Comment:
- to_use -> value?
- why do we have a `name` arg?
- seems like we can prefix this func with `_`